File: app/src/main/java/com/example/networkaplication/home/adapter/ListPresenter.java
package com.example.networkaplication.home.adapter;

import java.util.ArrayList;

public class ListPresenter {
    private final ArrayList<ItemData> items;

    public ListPresenter(ArrayList<ItemData> items) { this.items = items; }

    public void onBindRowViewAtPosition(RowView rowView, int position) {
        ItemData item = items.get(position);
        rowView.setImage(item.getImageUrl());
        rowView.setTitle(item.getTitle());
    }

    public int getRowsCount() {
        return items.size();
    }

    public ItemData get(int position) {
        return items.get(position);
    }
}

File: app/src/main/java/com/example/networkaplication/home/adapter/ListPresenter.kt
package com.example.networkaplication.home.adapter

import java.util.ArrayList

class ListPresenter(private val items: ArrayList<ItemData>) {

    val rowsCount: Int
        get() = items.size

    fun onBindRowViewAtPosition(rowView: RowView, position: Int) {
        val item = items[position]
        rowView.setImage(item.imageUrl!!)
        rowView.setTitle(item.title!!)
    }

    operator fun get(position: Int): ItemData {
        return items[position]
    }
}
